Rename Indexer client to Generator client throughout documentation (#253)
shinzonetwork/docs: content/reference/architecture-overview/index.md

  User -. "pay / subscribe" .-> SH
{% end %}

> The scheduler and gateway are coordination services. They match generator to hosts and route user queries to the right host. They never touch the actual data.

Reviewer: ```suggestion
The scheduler and gateway are coordination services. They match the Generator client to Host clients and route user queries to the right host. They never touch the actual data.
```

## Component inventory

| Component | Layer | Language | Repo | Touches data? |
| --- | --- | --- | --- | --- |
| Geth (source chain node) | Off-chain | Go | [go-ethereum](https://github.com/ethereum/go-ethereum) | Yes |

> | Generator client | Off-chain | Go | [shinzo-generator-client](https://github.com/shinzonetwork/shinzo-generator-client/) | Yes |

Reviewer 2: note: the repo is not yet renamed, should wait for it

Author: yes, i'll co-ordinate with team and then we will merge this PR

| --- | --- | --- |
| `0x0210` | View Registry | Registers views, deploys SVS-1 contracts. |
| `0x0211` | Host Registry | Tracks registered hosts. |

> | `0x0212` | Generator Registry | Tracks registered generator. |

Reviewer 2: note: lost the plural form

> | `0x0212` | Generator Registry | Tracks registered generator. |

Reviewer: ```suggestion
| `0x0212` | Generator Registry | Tracks registered Generator clients. |
```

SourceHub manages two independent policy domains:

> 1. Protocol participation: which DIDs are registered as hosts or generator (`group:host`, `group:generator`).

Reviewer: ```suggestion
1. Protocol participation: which DIDs are registered as Host clients or Generator clients (`group:host`, `group:generator`).
```

Outpost contracts are deployed on each source chain. They handle two things:

> - Validator assertions: a validator proves their identity using chain-native mechanisms and authorises an operator (delegate) key to act as their generator. On Ethereum, the validator's withdrawal key signs an EIP-712 message on the outpost contract that names the validator (consensus public key) and the operator pubkey. The contract verifies the signature, checks validator status, and emits an `AssertionSigned` event.

Reviewer: ```suggestion
- Validator assertions: a validator proves their identity using chain-native mechanisms and authorises an operator (delegate) key to act as their Generator. On Ethereum, the validator's withdrawal key signs an EIP-712 message on the outpost contract that names the validator (consensus public key) and the operator pubkey. The contract verifies the signature, checks validator status, and emits an `AssertionSigned` event.
```

### DefraDB

> DefraDB is the peer-to-peer document database embedded in every generator and host. Three separate DefraDB instances exist in the system (one per generator, one per host for primitives, one per host for view data).

Reviewer: ```suggestion
DefraDB is the peer-to-peer document database embedded in every Generator client and Host client. Three separate DefraDB instances exist in the system (one per Generator, one per Host for primitives, one per Host for view data).
```

Data replication uses libp2p, managed internally by DefraDB. The flow for a new document:

> 1. Generator writes document to its local DefraDB.

Reviewer: ```suggestion
1. Generator client writes document to its local DefraDB.
```

1. DefraDB computes a content digest.
1. DefraDB gossips the digest to connected peers.
1. Peers that want the document request the full content.
1. DefraDB sends the full document.

> Generators are write-only. They publish documents and reject all incoming replication (enforced by a replication filter in the Generator client). Hosts accept incoming documents from generator and replicate attestation records between each other.

Reviewer: ```suggestion
Generator clients are write-only. They publish documents and reject all incoming replication (enforced by a replication filter in the Generator client). Host clients accept incoming documents from generator and replicate attestation records between each other.
```

> Peers discover each other through bootstrap peers configured in DefraDB and through `EntityRegistered` events from ShinzoHub (when new generator or hosts join the network).

Reviewer: ```suggestion
Peers discover each other through bootstrap peers configured in DefraDB and through `EntityRegistered` events from ShinzoHub (when a new Generator client or Host client joins the network).
```

### Signing and attestation

> Each generator signs every block batch it produces. After writing all documents for a block (Block, Transaction, Log, AccessListEntry), the generator computes a Merkle root over all document CIDs, signs it with its identity key, and writes a `BlockSignature` document. 

Reviewer: ```suggestion
Each Generator client signs every block batch it produces. After writing all documents for a block (Block, Transaction, Log, AccessListEntry), the Generator client computes a Merkle root over all document CIDs, signs it with its identity key, and writes a `BlockSignature` document. 
```

> Hosts verify the signature included in the `blockSignature` against the CID for that document. They can then create an `AttestationRecord` (document) for that block to track how many independent generator produced the same data. The `vote_count` field uses a P-counter CRDT, which tracks the number of generator which produced that block. 

Reviewer: ```suggestion
Hosts verify the signature included in the `blockSignature` against the CID for that document. They can then create an `AttestationRecord` (document) for that block to track how many independent Generator clients produced the same data. The `vote_count` field uses a P-counter CRDT, which tracks the number of Generator clients which produced that block. 
```

> Snapshots bundle multiple blocks into signed files for faster initial sync. The generator periodically creates `SnapshotSignature` documents whose Merkle root is computed over the per-block `BlockSignature` roots within the range. So you get a two-level Merkle tree: document CIDs roll up into per-block roots, which roll up into per-snapshot roots.

Reviewer: ```suggestion
Snapshots bundle multiple blocks into signed files for faster initial sync. The Generator client periodically creates `SnapshotSignature` documents whose Merkle root is computed over the per-block `BlockSignature` roots within the range. So you get a two-level Merkle tree: document CIDs roll up into per-block roots, which roll up into per-snapshot roots.
```

## Registration flows

### Generator registration

> Spans two chains and a relayer. The generator cannot register on ShinzoHub until the validator has signed an assertion that ties the generator's operator key to the validator's identity.

Reviewer: ```suggestion
Spans two chains and a relayer. The Generator cannot register on ShinzoHub until the validator has signed an assertion that ties the Generator client's operator key to the validator's identity.
```
